src/Shared/BiometryService.iOS.cs

```diff
@@ -0,0 +1,414 @@
+			}
+			throw;
+		}
+	}
+
+	/// <inheritdoc/>
+	public void Remove(string keyName)
+	{
+		if (_logger.IsEnabled(LogLevel.Debug))
+		{
+			_logger.LogDebug("Removing the key '{keyName}'.", keyName);
+		}
+
+		var record = new SecRecord(SecKind.GenericPassword)
+		{
+			Service = keyName.ToLowerInvariant(),
+			UseOperationPrompt = _useOperationPrompt
+		};
+
+		var status = SecKeyChain.Remove(record);
+		if (status is not SecStatusCode.Success)
+		{
+			if (_logger.IsEnabled(LogLevel.Debug))
+			{
+				_logger.LogDebug("The key '{keyName}' has not been successfully removed.", keyName);
+			}
+			throw new BiometryException(BiometryExceptionReason.Failed, $"Something went wrong while removing the key '{keyName}'. Status = {status}");
+		}
+
+		if (_logger.IsEnabled(LogLevel.Debug))
+		{
+			_logger.LogDebug("The key '{keyName}' has been successfully removed.", keyName);
+		}
+	}
+
+	/// <summary>
+	/// Gets <see cref="BiometryType"/> from <see cref="LABiometryType"/>.
+	/// </summary>
+	/// <param name="biometryType"><see cref="LABiometryType"/>.</param>
+	/// <returns><see cref="BiometryType"/>.</returns>
+	private static BiometryType GetBiometryTypeFrom(LABiometryType biometryType)
+	{
+		return biometryType switch
+		{
+			LABiometryType.None => BiometryType.None,
+			LABiometryType.TouchId => BiometryType.Fingerprint,
+			LABiometryType.FaceId => BiometryType.Face,
+			_ => BiometryType.None,
+		};
+	}
+
+	/// <summary>
+	/// Encrypt a key using biometry.
+	/// </summary>
+	/// <remarks>
+	/// If the key already exists, it will be replaced.
+	/// </remarks>
+	/// <param name="keyName">The key name.</param>
+	/// <param name="keyValue">The key value.</param>
+	/// <exception cref="BiometryException">.</exception>
+	private void SetValueForKey(string keyName, string keyValue)
+	{
+		if (_logger.IsEnabled(LogLevel.Debug))
+		{
+			_logger.LogDebug("Saving the key '{keyName}'.", keyName);
+		}
+
+		var record = new SecRecord(SecKind.GenericPassword)
+		{
+			Service = keyName.ToLowerInvariant(),
+		};
+
+		// Check for duplicate key.
+		var status = SecKeyChain.Remove(record);
+		if (status is not SecStatusCode.Success & status is not SecStatusCode.ItemNotFound)
+		{
+			throw new BiometryException(BiometryExceptionReason.Failed, $"Something went wrong while checking for duplicate key '{keyName}'. Status = {status}");
+		}
+
+		// Use biometry to encrypt the key.
+		record.AccessControl = new SecAccessControl(
+			SecAccessible.WhenPasscodeSetThisDeviceOnly,
+			_fallbackOnPasscodeAuthentication ? SecAccessControlCreateFlags.UserPresence : SecAccessControlCreateFlags.BiometryCurrentSet
+		);
+
+		record.Generic = NSData.FromString(keyValue);
+
+		var result = SecKeyChain.Add(record);
+		if (result is not SecStatusCode.Success)
+		{
+			throw new BiometryException(BiometryExceptionReason.Failed, $"Something went wrong while saving the key '{keyName}'. Status = {status}");
```

Comment:
is status still releveent here since it's only calculated when removing the duplicated key

Reply:
Well it's a bit of information that could may be tell us why it failed but if you think I should remove it, I can.